### DISCUSSION TITLE
How to add a rectangle and some text on the GpuMat image?

### DISCUSSION THREAD

**Discussion by riyasharma15** [emgucv/emgucv#967] [Q&A]
I was trying to add a rectangle and some text on the GpuMat image but the CudaInvoke class that we have for GpuMat does not have any Rectangle and PutText methods. 

So is there any way to add a rectangle and some text on a GpuMat image? Or are there any libraries that would let me do this on a GpuMat image?

**Comment by sn4k3** [comment]
`GpuMat` can only be used with `GpuInvoke`, all other operations not present under it will require a conversion to `Mat` and `CvInvoke` which will kill the performance.
Stay with `Mat` unless you do lot of sequential supported operations by Cuda
